traitsui/wx/range_editor.py

```diff
@@ -244,6 +244,9 @@ def update_object_on_enter(self, event):
                 # Pretend it didn't happen (i.e. do not change self.value).
                 value = self.value
                 self.control.text.SetValue(str(value))
+                # for compound editor, value may be non-numeric
+                if not isinstance(value, (int, float)):
```

Comment:
This fails with long ints on Python 2.7 (which we still have to support at least until the end of the year) as well as likely other numeric types which might be reasonable (eg. NumPy scalar types, standard lib `Decimal` and `Rational`).

Additionally, we want to run as much of the subsequent code as we can, since we want to reset colors and error handling if the value is valid for the underlying trait.  I think a better approach is to wrap the `self.control.slider.SetValue` in a `try...except...` ("it's better to ask forgiveness" in this case, as opposed to "look before you leap")

Reply:
I have made the changes for wx.
For QT, the code looks for
`if not isinstance(value, (int, float)): ...
`
Maybe this should be changed as well?